python/triqs/gf/tools.py

```diff
@@ -259,8 +259,94 @@ def fit_legendre(g_t, order=10):
     l = np.arange(len(lmesh))
     scale = np.sqrt(2.*l + 1) / mesh.beta
     scale = scale.reshape([len(lmesh)] + [1]*len(g_t.target_shape))
-    
+
     g_l = Gf(mesh=lmesh, target_shape=g_t.target_shape)
     g_l.data[:] = c_l.reshape(g_l.data.shape) / scale
 
     return g_l
+
+
+def discretize_bath(delta_inp, Nb , bandwidth = 3, t_val = 0.0, tol=1e-5):
+    """
+    discretizes a given Delta_iw with Nb bath sites
+
+    CAUTION: This function works only for 1x1 / scalar valued Gf objects.
+    if necessary call for each orbital in input Delta_iw separately.
+
+    Parameters
+    ----------
+    g : GfImFreq
+        Non-interacting Green's function.
+    Nb : int
+        number of bath sites
+    bandwidth : approximative bandwidth for init of bath energies
+    t_val : float
+        to which value should the bath hoppings be initialized
+    tol: float
+        tolerance for scipy minimize
```

Comment:
```suggestion
    bandwidth : float, optional
        approximate bandwidth, used in choice of initial of bath energies [default=3.0]
    t_init : float, optional
        initial guess used for all hopping values [default=0.0]
    tol: float, optional
        tolerance for scipy minimize [default=1e-5]
```

Also we may want to use a more conservative default for the tolerance (~1e-8)?

Reply:
I checked, and scipy optimize as problems for that small tolerances. Sometimes 1e-8 will fail. We could maybe go for 1e-7 or 1e-6 for now. Will change that.